modules/provision/crypto.go

```diff
@@ -0,0 +1,101 @@
+package provision
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"fmt"
+	"io"
+
+	"github.com/pkg/errors"
+	"github.com/threefoldtech/zosv2/modules/identity"
+
+	"github.com/threefoldtech/zosv2/modules/crypto"
+	"golang.org/x/crypto/ed25519"
+)
+
+// Sign creates a signature from all the field of the reservation
+// object and fill the Signature field
+func (r *Reservation) Sign(privateKey ed25519.PrivateKey) error {
+	buf := &bytes.Buffer{}
+	_, err := buf.WriteString(r.ID)
+	if err != nil {
+		return err
+	}
+	_, err = buf.WriteString(r.User)
+	if err != nil {
+		return err
+	}
+	_, err = buf.WriteString(string(r.Type))
+	if err != nil {
+		return err
+	}
+	_, err = buf.Write(r.Data)
+	if err != nil {
+		return err
+	}
+
+	signature, err := crypto.Sign(privateKey, buf.Bytes())
+	if err != nil {
+		return err
+	}
+	r.Signature = signature
+	return nil
+}
+
+// Verify verifies the signature of the reservation
+func Verify(r Reservation) error {
```

Comment:
Don't see a reason why this can't be an instance method.

```suggestion
func (r *Reservation) Verify() error {
```

Reply:
was just to not overload the API of the Reservation struct. could be change to a method